litestar/plugins/base.py

```diff
@@ -267,15 +269,20 @@ def __init__(self, plugins: list[PluginProtocol]) -> None:
         self.serialization = tuple(p for p in plugins if isinstance(p, SerializationPluginProtocol))
         self.cli = tuple(p for p in plugins if isinstance(p, CLIPluginProtocol))
 
-    def get(self, type_: type[PluginT]) -> PluginT:
+    def get(self, type_: type[PluginT] | str) -> PluginT:
```

Comment:
I don’t think this belongs in the `get` method since it’s conceptually different: Right now the registry is immutable and `get` just returns plugins that were previously registered I straw if loading them.

If we want to allow retrieval via string, it should just look up e.g. the `__name__` attribute of the types of the plugins already registered. If we want to allow *loading* plugins via string, this should happen on app init.

Reply:
Fair point.  Are you ok with me leaving the utils that I've added here?  They were originally in Litestar but were removed at some point.

Reply:
> Are you ok with me leaving the utils that I've added here? They were originally in Litestar but were removed at some point.

Sure. We could probably also reuse those in the CLI for the autoimports at some point.